src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java

```diff
@@ -725,23 +503,36 @@ private AIState startBuilding()
             onStartWithoutStructure();
             return AIState.IDLE;
         }
+        final AIState nextState;
         switch (currentStructure.getStage())
         {
             case CLEAR:
```

Comment:
Nope, wrong solution, I object. 

Sonar is not a sentient being. We are.

So we change this back and adapt the sonar rules to match what we want.